docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,596 @@
+least 16 bytes.
+
+## `@_alwaysEmitIntoClient`
+
+Forces the body of a function to be emitted into client code.
+
+Note that this is distinct from `@inline(__always)`; it doesn't force inlining
+at call-sites, it only means that the implementation is compiled into the
+module which uses the code.
+
+This means that `@_alwaysEmitIntoClient` definitions are _not_ part of the
+defining module's ABI, so changing the implementation at a later stage
+does not break ABI.
+
+Most notably, default argument expressions are implicitly
+`@_alwaysEmitIntoClient`, which means that adding a default argument to a
+function which did not have one previously does not break ABI.
+
+## `@_borrowed`
+
+Indicates that the [conservative access pattern](/docs/Lexicon.md#access-pattern)
+for some storage (a subscript or a property) should use the `_read` accessor
+instead of `get`.
+
+For more details, see the forum post on
+[Value ownership when reading from a storage declaration](https://forums.swift.org/t/value-ownership-when-reading-from-a-storage-declaration/15076).
+
+## `@_cdecl("cName")`
+
+Similar to `@_silgen_name` but uses the C calling convention.
+
+This attribute doesn't have very well-defined semantics. Type bridging is not
+done, so the parameter and return types should correspond directly to types
+accessible in C. In most cases, it is preferable to define a static method
+on an `@objc` class instead of using `@_cdecl`.
+
+For potential ideas on stabilization, see
+[Formalizing `@cdecl`](https://forums.swift.org/t/formalizing-cdecl/40677).
+
+## `@_disfavoredOverload`
+
+Marks an overload that the type checker should try to avoid using. When the
+expression type checker is considering overloads, it will prefer a solution
+with fewer `@_disfavoredOverload` declarations over one with more of them.
+
+Historically, it was designed to help with `ExpressibleByXYZLiteral` types.
+The type checker strongly prefers to give literals their default type
+(e.g. `Int` for `ExpressibleByIntegerLiteral`,
+`String` for `ExpressibleByStringLiteral`, etc.).
+If an API should prefer some other type, but accept the default too,
+marking the declaration taking the default type with `@_disfavoredOverload`
+gives the desired behavior:
+
+```swift
+extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
+
+extension Text {
+  // We want `Text("foo")` to use this initializer:
+  init(_ key: LocalizedStringKey) { ... }
+
+  // But without @_disfavoredOverload, it would use this one instead,
+  // because that lets it give the literal its default type:
+  @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
+}
+```
+
+`@_disfavoredOverload` is a stop-gap design; eventually the type-checker
+behavior should be fixed and the attribute should be removed/made into a no-op.
+
+Its main use case is to work around clearly undesirable overload resolution,
+not to arbitrarily manipulate overload priority.
```

Comment:
I'm going to suggest changing the wording here to emphasize its role as a usage guideline:

> Use `@_disfavoredOverload` to work around known bugs in the overload resolution rules that cannot be immediately fixed without a source break. Don't use it to adjust overload resolution rules that are otherwise sensible but happen to produce undesirable results for your particular API; it will likely be removed or made into a no-op eventually, and then you will be stuck with an overload set that cannot be made to function in the way you intend.
> 
> `@_disfavoredOverload` was first introduced to work around a bug in overload resolution with `ExpressibleByXYZLiteral` types. The type checker strongly prefers to give literals their default type (e.g. `Int` for `ExpressibleByIntegerLiteral`, `String` for `ExpressibleByStringLiteral`, etc.). If an API should prefer some other type, but accept the default too, marking the declaration taking the default type with `@_disfavoredOverload` gives the desired behavior:
> 
>```swift
> extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
> 
> extension Text {
>   // We want `Text("foo")` to use this initializer:
>   init(_ key: LocalizedStringKey) { ... }
> 
>   // But without @_disfavoredOverload, it would use this one instead,
>   // because that lets it give the literal its default type:
>   @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
> }
>```

`````
Use `@_disfavoredOverload` to work around known bugs in the overload
resolution rules that cannot be immediately fixed without a source break.
Don't use it to adjust overload resolution rules that are otherwise sensible
but happen to produce undesirable results for your particular API; it will
likely be removed or made into a no-op eventually, and then you will be
stuck with an overload set that cannot be made to function in the way
you intend.

`@_disfavoredOverload` was first introduced to work around a bug in overload
resolution with `ExpressibleByXYZLiteral` types. The type checker strongly
prefers to give literals their default type (e.g. `Int` for 
`ExpressibleByIntegerLiteral`, `String` for `ExpressibleByStringLiteral`,
etc.). If an API should prefer some other type, but accept the default too,
 marking the declaration taking the default type with `@_disfavoredOverload`
 gives the desired behavior:

```swift
extension LocalizedStringKey: ExpressibleByStringLiteral { ... }

extension Text {
  // We want `Text("foo")` to use this initializer:
  init(_ key: LocalizedStringKey) { ... }

  // But without @_disfavoredOverload, it would use this one instead,
  // because that lets it give the literal its default type:
  @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
}
```
`````